homeassistant/components/portainer/__init__.py

```diff
@@ -0,0 +1,75 @@
+"""The Portainer integration."""
+
+from __future__ import annotations
+
+import logging
+
+from pyportainer import (
+    Portainer,
+    PortainerAuthenticationError,
+    PortainerConnectionError,
+    PortainerTimeoutError,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_HOST, CONF_PORT, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import (
+    ConfigEntryAuthFailed,
+    ConfigEntryError,
+    ConfigEntryNotReady,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .coordinator import PortainerCoordinator
+from .models import PortainerData
+
+_LOGGER = logging.getLogger(__name__)
+
+_PLATFORMS: list[Platform] = [Platform.BINARY_SENSOR]
+
+type PortainerConfigEntry = ConfigEntry[PortainerData]
```

Comment:
you can also move this to `coordinator.py`
```suggestion
type PortainerConfigEntry = ConfigEntry[PortainerCoordinator]
```